src/assets/scripts/templates/customers/addresses.js

```diff
@@ -8,31 +8,37 @@
  */
 
 import $ from 'jquery';
+import {CountryProvinceSelector} from '@shopify/theme-addresses';
 
 const $newAddressForm = $('#AddressNewForm');
 
 if ($newAddressForm.length) {
-  // Initialize observers on address selectors, defined in shopify_common.js
-  if (Shopify) {
-    new Shopify.CountryProvinceSelector(
-      'AddressCountryNew',
-      'AddressProvinceNew',
-      {
-        hideElement: 'AddressProvinceContainerNew',
-      },
-    );
-  }
+  const countryProvinceSelector = new CountryProvinceSelector(window.theme.allCountryOptionTags);
+
+  countryProvinceSelector.build(
+    $('#AddressCountryNew')[0],
```

Comment:
Also we tend to use data attributes to make JS DOM selections. Can we switch to that?

Reply:
No, we cannot use data attribute to auto detect country and province pairs. We need to bound a `change` event of a country select to a province selector. There is no guarantee that getting a query list of elements based on a data attribute would give us an ordered list pairs that we need to associate because this highly depends on the layout of the html.